hibernate-core/src/main/java/org/hibernate/resource/jdbc/internal/ResourceRegistryStandardImpl.java

```diff
@@ -201,13 +205,15 @@ public void register(ResultSet resultSet, Statement statement) {
 			}
 		}
 		if ( statement != null ) {
+			Set<ResultSet> resultSets = xref.get( statement );
+
 			// Keep this at DEBUG level, rather than warn.  Numerous connection pool implementations can return a
 			// proxy/wrapper around the JDBC Statement, causing excessive logging here.  See HHH-8210.
-			if ( log.isDebugEnabled() && !xref.containsKey( statement ) ) {
```

Comment:
Same here, this optimization will only change anything when the `DEBUG` level is enabled.